tests/models/test_energy_plan.py

```diff
@@ -0,0 +1,249 @@
+        )
+        self.electricity_plan_night_uncontrolled_controlled = ElectricityPlan(
+            name="UncontrolledPlan",
+            daily_charge=1.0,
+            nzd_per_kwh={
+                "Night": night,
+                "Uncontrolled": uncontrolled,
+                "Controlled": controlled,
+            },
+        )
+        self.electricity_plan_night_uncontrolled = ElectricityPlan(
+            name="UncontrolledPlan",
+            daily_charge=1.0,
+            nzd_per_kwh={"Night": night, "Uncontrolled": uncontrolled},
+        )
+
+    def test_all_inclusive_plan(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"All inclusive"}.
+        """
+        cost = self.electricity_plan_all_inclusive.calculate_cost(self.profile)
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, (300 + 100) * 0.18))
+
+    def test_day_night_plan(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"Day", "Night"}.
+        """
+        cost = self.electricity_plan_day_night.calculate_cost(self.profile)
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, 300 * 0.25 + 100 * 0.15))
+
+    def test_uncontrolled(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"Uncontrolled"}.
+        """
+        cost = self.electricity_plan_uncontrolled.calculate_cost(self.profile)
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, (300 + 100) * 0.22))
+
+    def test_uncontrolled_controlled(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"Uncontrolled", "Controlled"}.
+        """
+        cost = self.electricity_plan_uncontrolled_controlled.calculate_cost(
+            self.profile
+        )
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, 300 * 0.22 + 100 * 0.20))
+
+    def test_uncontrolled_all_inclusive(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"Uncontrolled", "All inclusive"}.
+        """
+        cost = self.electricity_plan_uncontrolled_all_inclusive.calculate_cost(
+            self.profile
+        )
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, (300 + 100) * 0.18))
+
+    def test_night_controlled_day(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"Night", "Controlled", "Day"}.
+        """
+        cost = self.electricity_plan_night_controlled_day.calculate_cost(self.profile)
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, 300 * 0.25 + 100 * 0.15))
+
+    def test_night_all_inclusive(self):
+        """
+        Test electricity plan with variable pricing pattern
+        {"Night", "All inclusive"}.
+        """
+        cost = self.electricity_plan_night_all_inclusive.calculate_cost(self.profile)
+        self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, 300 * 0.18 + 100 * 0.15))
+
+    def test_night_uncontrolled_controlled(self):
+        """
```

Comment:
Shouldn't this need the Controlled value to be included?
 
  def test_night_uncontrolled_controlled(self) 
  
 self.assertEqual(cost, (1.0 * DAYS_IN_YEAR, 300 * 0.22 + 100 * 0.15))

Reply:
Yes, it is being used but it's a magic number. Will push up an improvement to the test to make it clearer.